common/app/routes/Challenges/views/Modern/Show.jsx

```diff
@@ -39,23 +47,30 @@ export const mapStateToPanes = addNS(
   createSelector(
     filesSelector,
     showPreviewSelector,
-    (files, showPreview)=> {
+    (files, showPreview) => {
       // create panes map here
       // must include map
       // side panel
       // editors are created based on state
       // so pane component can have multiple panes based on state
-      const panesMap = _.reduce(files, (map, file) => {
-        map[createModernEditorToggleType(file.fileKey)] = file.name;
-        return map;
-      }, {
+
+      let panesMap = {
         [types.toggleMap]: 'Map',
-        [types.toggleSidePanel]: 'Lesson'
-      });
+        [types.toggleSidePanel]: 'Lesson',
+        [createModernEditorToggleType(getFirstFileKey(files))]: 'Editor'
+      };
+
+      // If there is more than one file show file name
+      if (Object.keys(files).length > 1) {
+        _.forEach(files, (file) => {
+          panesMap[createModernEditorToggleType(file.fileKey)] = file.name;
+        });
+      }
```

Comment:
This needs to be an if/else

When there is more then one file there shouldn't be an `Editor` plan, since there will be multiple.